simple_history/models.py

```diff
@@ -275,11 +276,14 @@ def get_history_user(self, instance):
             return instance._history_user
         except AttributeError:
             try:
-                if self.thread.request.user.is_authenticated():
-                    return self.thread.request.user
-                return None
+                is_authenticated = self.thread.request.user.is_authenticated
             except AttributeError:
                 return None
+            if not is_authenticated in (True, False):
+                is_authenticated = is_authenticated() # Django < 1.10
```

Comment:
I think this is not correct.  In 1.10 or 1.11, `is_authenticated` is an instance of CallableTrue or CallableFalse, not True or False, so the `if not in` test will fail and go into line 283 which will raise the deprecation warning.

Reply:
Yes, that might be. I tested it having the callable boolean in mind but now I saw that none of the implemented tests get to this line at all in Django 1.10+ (because `self.thread.request.user.is_authenticated` raises the caught `AttributeError`.

`request.user` raises an `AttributeError` in multiple tests in Django 2.0 (`AttributeError: 'WSGIRequest' object has no attribute 'user'`). For that reason it might be the next step to fix those.

Reply:
This error might be caused by a wrong ordering of middleware.